src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -109,24 +108,28 @@ String metricName(MetricName metricName) {
         return prefix + '_' + group + '_' + name;
     }
 
-    static MetricFamilySamples convert(String name, String help, KafkaMetric metric, Map<String, String> labels) {
-        Object value = metric.metricValue();
-        if (!(value instanceof Number)) {
-            // Prometheus only accepts numeric metrics.
-            // Kafka gauges can have arbitrary types, so skip them for now
-            // TODO move non-numeric values to labels
-            return null;
-        }
-        Map<String, String> sanitizedLabels = labels.entrySet().stream()
+    private static MetricFamilySamples convert(String prometheusMetricName, KafkaMetric metric, MetricName metricName) {
+        Map<String, String> sanitizedLabels = metricName.tags().entrySet().stream()
                 .collect(Collectors.toMap(
                         e -> Collector.sanitizeMetricName(e.getKey()),
                         Map.Entry::getValue,
                         (v1, v2) -> {
-                            throw new IllegalStateException("Unexpected duplicate key " + v1);
+                            LOG.warn("Ignoring metric value duplicate key {}", v1);
+                            return v1;
                         },
                         LinkedHashMap::new));
```

Comment:
It seems we have the exact same block in YammerMetricsCollector. Maybe we can extract this into a method. WDYT?

Reply:
Using a helper method of sorts? Where would be the best place to have the extracted method that would be called from both Classes?

Reply:
While not perfect, `MetricFamilySamplesBuilder` could do.

Reply:
Sorry, can you elaborate on `MetricFamilySamplesBuilder could do`.

Reply:
I meant we could put that new method in `MetricFamilySamplesBuilder`

Reply:
New `sanitizeLabels` method added to `MetricFamilySamplesBuilder` and called in `convert` methods